### PR TITLE
Added English Tutorial for LOJ-1159 Batman(en) 

### PR DESCRIPTION
**Pull request by Tahsin-Zaman-Jilan**
https://lightoj.com/problem/batman

### CODE REVIEW

File: 1159/en.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+The function batman takes in three strings, X, Y, and Z, as well as their lengths, and returns the length of the LCS of the three strings. 
+It does this by filling in a 3-dimensional array, L, using a bottom-up dynamic programming approach.
+
+The base case is when either the length of X, Y, or Z is 0, in which case the LCS is 0. If the last characters of X, Y, and Z are the same,
+then the LCS of the three strings is one more than the LCS of the three strings without the last characters. 
+Otherwise, the LCS is the maximum of the LCS of the three strings with X, Y, or Z shortened by one character. 
+# In short 
+-if the string matches then it goes diagoanlly(add 1 and increase i,j,k index)
+-if not then take max of neighbour elements
```
Comment (rebornplusplus) [inline 1159/en.md:15]: This is just explaining what your code is doing. This won't fly.
Explain what the solution is, using sentences so that the reader doesn't need to resort to your source code to understand what's happening. 
Since this is a DP problem, you can start off by explaining what the states are, how computing the result for one state depends on multiple small states (in short, explain the sub-problems). 
As you mentioned LCS, it'd be good of you to provide a reference to an article about LCS as well; so that if the reader hasn't encountered LCS before, they can quickly jump to that article, read it and try the problem themselves again, before reading your editorial.

File: 1159/en.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	int m = X.length();
+	int n = Y.length();
+	int o = Z.length();
+
+	cout << "Case "<< idx<<": "<< batman(X, Y,Z, m, n, o)<<endl;;
+
+	
+}
+}
+```
```
Comment (rebornplusplus) [inline 1159/en.md:77]: The source code is poorly indented.

File: 1159/en.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+
+# LOJ 1159
```
Comment (rebornplusplus) [inline 1159/en.md:2]: Add the problem's name as well, next to the number.

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Good effort. But a few things need to be addressed before this goes in.

File: 1159/en.md (hunk 1 above)
Comment (Tahsin-Zaman-Jilan) [reply 1159/en.md:15]: Ok i am working on that

File: 1159/en.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+
+# [LOJ 1159 Batman](https://lightoj.com/problem/batman)
+
+## Solution Approach:
+In this problem given 3 string and need to find the length of the longest common subsequence 
+[LCS](https://www.programiz.com/dsa/longest-common-subsequence) of these three strings.
+
+Its not necessary that the characters have to be next to each other 
+in the original strings.
```
Comment (rebornplusplus) [inline 1159/en.md:9]: Confusing line. Unnecessary. I would remove it, since sub-sequence by itself means that, but more definitively.

File: 1159/en.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+One approach to solve the problem is to create a 3-dimensional table with dimensions corresponding
+to the lengths of each of the three input strings.
+The sub-problems in this case are the substrings 
+dp[1...i], Y[1...j], Z[1...k] for all possible values of i, j, k
```
Comment (rebornplusplus) [inline 1159/en.md:14]: I would rewrite this as:

> Let's define LCS[i][j][k] to be the LCS of the three sub-strings X[1..i], Y[1..j], Z[1..k].

File: 1159/en.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+To compute the value of LCS[i][j][k], we consider three possibilities:
+- You can start by checking base case whether the length of any string is 0, in which case the LCS is 0.
+- if three characters match, length of the common subsequence would be 1 plus 
+  the length of the common subsequence till the i-1,j-1 and k-1 indexes
```
Comment (rebornplusplus) [inline 1159/en.md:19]: > if three characters match

More like,

> if the last characters of each string match

File: 1159/en.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+- Otherwise, the LCS is will be maximum of value of the element above or to the left of the 
+  current element of three strings(max of neighbour elements).
```
Comment (rebornplusplus) [inline 1159/en.md:21]: I would rephrase it. Something like:

> Otherwise, we can drop the last character of any one of the three strings and take the maximum value i.e. `LCS[i][j][k] = max(LCS[i-1][j][k], LCS[i][j-1][k], LCS[i][j][k-1])`.

File: 1159/en.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+The final LCS value is stored in LCS[m][n][o], which is the LCS of the three input strings.
```
Comment (rebornplusplus) [inline 1159/en.md:23]: `m`, `n`, `o` are undefined. Write `len(X)`, `len(Y)` and `len(Z)` instead.

File: 1159/en.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+This question is similar to [this question](https://leetcode.com/problems/longest-common-subsequence/description/), you can check that also.
```
Comment (rebornplusplus) [inline 1159/en.md:25]: ~~question~~ problem

File: 1159/en.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+	int m = X.length();
+	int n = Y.length();
+	int o = Z.length();
+
+	cout << "Case "<< idx<<": "<< LCS(X, Y,Z, m, n, o)<<endl;
+
+	
+}
+}
+```
```
Comment (rebornplusplus) [inline 1159/en.md:82]: It is still not properly indented.

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Good changes. However it still needs improvement.

**Comment by Tahsin-Zaman-Jilan** [comment]
@rebornplusplus sir can please check now?

**Review by rebornplusplus — APPROVED** [review]
Looks good.

The indentation was still bad, so I added a commit to fix it.
